tests/std/tests/P2446R2_views_as_rvalue/test.cpp

```diff
@@ -0,0 +1,453 @@
+
+            STATIC_ASSERT(same_as<decltype(views::as_rvalue(std::move(rng))), VS>);
+            STATIC_ASSERT(noexcept(views::as_rvalue(std::move(rng))) == is_noexcept);
+
+            STATIC_ASSERT(same_as<decltype(std::move(rng) | views::as_rvalue), VS>);
+            STATIC_ASSERT(noexcept(std::move(rng) | views::as_rvalue) == is_noexcept);
+        }
+
+        // ... with const rvalue argument
+        STATIC_ASSERT(CanViewAsRvalue<const remove_reference_t<Rng>> == (is_view && copy_constructible<V>) );
+        if constexpr (CanViewAsRvalue<const remove_reference_t<Rng>>) {
+            constexpr bool is_noexcept = is_nothrow_copy_constructible_v<V>;
+
+            STATIC_ASSERT(same_as<decltype(views::as_rvalue(std::move(as_const(rng)))), V>);
+            STATIC_ASSERT(noexcept(views::as_rvalue(std::move(as_const(rng)))) == is_noexcept);
+
+            STATIC_ASSERT(same_as<decltype(std::move(as_const(rng)) | views::as_rvalue), V>);
+            STATIC_ASSERT(noexcept(std::move(as_const(rng)) | views::as_rvalue) == is_noexcept);
+        }
+    }
+
+    // Validate deduction guide
+    same_as<R> auto r = as_rvalue_view{std::forward<Rng>(rng)};
+
+    // Validate as_const_view::size
+    STATIC_ASSERT(CanMemberSize<R> == ranges::sized_range<V>);
+    if constexpr (CanMemberSize<R>) {
+        same_as<ranges::range_size_t<V>> auto s = r.size();
+        assert(_To_unsigned_like(s) == ranges::size(expected));
+        STATIC_ASSERT(noexcept(r.size()) == noexcept(ranges::size(as_const(rng))));
+    }
+
+    // Validate as_const_view::size (const)
+    STATIC_ASSERT(CanMemberSize<const R> == ranges::sized_range<const V>);
+    if constexpr (CanMemberSize<const R>) {
+        same_as<ranges::range_size_t<const V>> auto s = as_const(r).size();
+        assert(_To_unsigned_like(s) == ranges::size(expected));
+        STATIC_ASSERT(noexcept(as_const(r).size()) == noexcept(ranges::size(rng)));
+    }
+
+    const bool is_empty = ranges::empty(expected);
+
+    // Validate view_interface::empty and operator bool
+    STATIC_ASSERT(CanMemberEmpty<R> == (forward_range<V> || ranges::sized_range<V>) );
+    STATIC_ASSERT(CanBool<R> == CanEmpty<R>);
+    if constexpr (CanMemberEmpty<R>) {
+        assert(r.empty() == is_empty);
+        assert(static_cast<bool>(r) == !is_empty);
+    }
+
+    // Validate view_interface::empty and operator bool (const)
+    STATIC_ASSERT(CanMemberEmpty<const R> == (forward_range<const Rng> || ranges::sized_range<const V>) );
+    STATIC_ASSERT(CanBool<const R> == CanEmpty<const R>);
+    if constexpr (CanMemberEmpty<const R>) {
+        assert(as_const(r).empty() == is_empty);
+        assert(static_cast<bool>(as_const(r)) == !is_empty);
+    }
+
+    assert(test_equal(r, expected)); // TRANSITION, GH-3009 (use ranges::equal)
+    if (!forward_range<V>) { // intentionally not if constexpr
+        return true;
+    }
+
+    // Validate as_rvalue_view::begin

```

Comment:
@CaseyCarter noted that we should verify that `begin()` and `end()` return `move_iterator` and/or `move_sentinel`; the test currently doesn't mention these types directly.